### PR TITLE
build: update to nodejs rules 0.41.0

### PR DESCRIPTION
**Pull request by gregmagolan**
This brings is changes to the `@nodejs` repository required for https://github.com/angular/angular/pull/33927. See release notes for more details: https://github.com/bazelbuild/rules_nodejs/releases/tag/0.41.0.

rules_nodejs is approaching 1.0 and breaking changes for that release are being made more frequently. In this release, the ts_devserver API changed and it no longer injects html script tags into a provided index.html file. The diff on this commit is large as this breaking change affects quite a few tests.

Also note that we don’t update @angular/bazel schematics and integration/bazel as 0.41.0 is not a recommended update for angular end-users yet due to the breaking changes in ts_devserver & web_package (now named pkg_web). When a suitable plain npm package that is in progress is finished then it will be possible to easily replace the html injection functionality removed from ts_devserver & pkg_web.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by gregmagolan** [comment]
Note: protractor data work-around in the PR can be removed after this https://github.com/bazelbuild/rules_nodejs/pull/1388 lands and makes it into the next release.

File: modules/playground/e2e_test/sourcemap/BUILD.bazel [hunk 1]
```diff
@@ -1,11 +1,24 @@
 load("//modules/playground/e2e_test:example_test.bzl", "example_test")
 
+filegroup(
+    name = "protractor_web_test_data_workaroud",
+    srcs = ["//modules/playground/src/sourcemap:index.ts"],
+)
+# Work-around for protracto_web_test data not allow files.
```
Comment (petebacondarwin) [inline modules/playground/e2e_test/sourcemap/BUILD.bazel]:
```suggestion
# Work-around for protractor_web_test data not allowing files.
```

**Review by petebacondarwin — APPROVED** [review]

File: modules/playground/src/web_workers/images/index.html [hunk 2]
```diff
@@ -1,8 +1,15 @@
 <html>
   <head>
     <link rel="stylesheet" href="image_demo.css" />
+    <link rel="stylesheet" href="/npm/node_modules/materialize-css/dist/css/materialize.min.css">
+    <link rel="stylesheet" href="/third_party/github.com/google/material-design-icons/material-icons.css"></head>
   </head>
   <body>
     <image-demo></image-demo>
+
+    <!-- <script src="/npm/node_modules/materialize-css/dist/js/materialize.min.js"></script>
```
Comment (josephperrott) [inline modules/playground/src/web_workers/images/index.html]: Is this still meant to be here and commented out?

**Review by josephperrott — APPROVED** [review]
LGTM

File: modules/playground/e2e_test/sourcemap/BUILD.bazel (hunk 1 above)
Comment (gregmagolan) [reply modules/playground/e2e_test/sourcemap/BUILD.bazel]: Thanks Pete

File: modules/playground/src/web_workers/images/index.html (hunk 2 above)
Comment (gregmagolan) [reply modules/playground/src/web_workers/images/index.html]: nope. good catch! thanks

**Review by josephperrott — APPROVED** [review]
LGTM, global approval

**Comment by josephperrott** [comment]
Caretaker note: Global Approval

**Pull request closed without merging**